api/src/wfl/module/copyfile.clj

```diff
@@ -14,57 +15,68 @@
   (cromwell/submit-workflow
     environment
     (util/extract-resource (:top workflow-wdl))
     nil
-    (-> workflow (select-keys [:src :dst]) (util/prefix-keys pipeline))
+    (-> inputs (util/prefix-keys pipeline))
     options
     labels))
 
-(defn add-copyfile-workload!
-  "Use transaction TX to add the workload described by WORKLOAD-REQUEST."
-  [tx {:keys [items] :as _workload-request}]
-  (let [workflow-options (-> (:cromwell _workload-request)
-                             all/cromwell-environments
-                             first
-                             util/make-options
-                             (util/deep-merge (:workflow_options _workload-request)))
-        [id table]       (all/add-workload-table! tx workflow-wdl _workload-request)
-        to-row (fn [item] (assoc (:inputs item)
-                            :workflow_options
-                            (json/write-str (util/deep-merge workflow-options (:workflow_options item)))))]
-    (letfn [(add-id [m id] (assoc m :id id))]
-      (jdbc/insert-multi! tx table (map add-id (map to-row items) (range))))
-    id))
+(defn create-copyfile-workload!
+  "Use transaction TX to add the workload described by REQUEST."
+  [tx {:keys [items common] :as request}]
+  (letfn [(merge-to-json [shared specific]
+            (json/write-str (util/deep-merge shared specific)))
+          (serialize [workflow id]
+            (-> workflow
+              (assoc :id id)
+              (update :inputs #(merge-to-json (:inputs common) %))
+              (update :options #(merge-to-json (:options common) %))))]
+    (let [[id table] (batch/add-workload-table! tx workflow-wdl request)]
+      (jdbc/insert-multi! tx table (map serialize items (range)))
+      (workloads/load-workload-for-id tx id))))
 
 (defn start-copyfile-workload!
   "Use transaction TX to start _WORKLOAD."
-  [tx {:keys [cromwell items uuid] :as workload}]
-  (let [env (first (all/cromwell-environments cromwell))]
-    (letfn [(submit! [{:keys [id inputs workflow_options]}]
-              [id (submit-workflow env inputs workflow_options {:workload uuid}) "Submitted"])
-            (update! [tx [id uuid status]]
+  [tx {:keys [items uuid] :as workload}]
+  (let [env             (get-cromwell-environment workload)
+        default-options (util/make-options env)]
+    (letfn [(submit! [{:keys [id inputs options]}]
+              [id (submit-workflow env inputs
+                    (util/deep-merge default-options options)
+                    {:workload uuid})])
+            (update! [tx [id uuid]]
               (jdbc/update! tx items
-                {:updated (OffsetDateTime/now) :uuid uuid :status status}
+                {:updated (OffsetDateTime/now) :uuid uuid :status "Submitted"}
                 ["id = ?" id]))]
       (run! (comp (partial update! tx) submit!) (:workflows workload))
       (jdbc/update! tx :workload
         {:started (OffsetDateTime/now)} ["uuid = ?" uuid]))))
```

Comment:
The implementations for `start-workload!` still differ somewhat. This is due to the xx module using (a faked) batch submit to cromwell. In a subsequent change, we could explore moving the other modules onto that endpoint and actually implement it. Given that, maybe we can extract common functionality for these modules (c.f. `create-workload!`)